test/integration/admin_dashboard_test.rb

```diff
@@ -388,6 +388,45 @@ def teardown
     assert_equal needle.title, advisor.theses.first.title
   end
 
+  # Department_Theses
+  test 'accessing department_thesis panel does not work with basic rights' do
+    mock_auth(users(:basic))
+    get '/admin/department_theses'
+    assert_response :redirect
+    mock_auth(users(:processor))
+    get '/admin/department_theses'
+    assert_response :redirect
+  end
+
+  test 'accessing department_thesis panel as an admin user works' do
+    mock_auth(users(:admin))
+    get '/admin/department_theses'
+    assert_response :success
+    assert_equal('/admin/department_theses', path)
+  end
+
+  test 'accessing department_thesis panel as a thesis_admin user works' do
+    mock_auth(users(:thesis_admin))
+    get '/admin/department_theses'
+    assert_response :success
+    assert_equal('/admin/department_theses', path)
+  end
+
+  test 'can edit department_thesis through admin dashboard' do
+    needle = false
```

Comment:
This definitely works, so can stay. However, I wanted to share the needle confused me when reading this and I'm not yet convinced (but am happy to be swayed) that it's clearer with the needle variable than just using `true` and `false` directly in the code. Looking at `!needle` and then having to look through to follow the state of `needle` to ensure it didn't change kinda broke my Friday afternoon brain lol

Reply:
Yeah, I can hear that feedback. I struggled a bit with whether I should use a variable for this. We use `needle` as a variable in tests several times throughout this file, but in this case it felt a bit awkward. Seeing your feedback on this makes me think that we should try it another way. Stay tuned...

Reply:
Yeah. To be clear I'm not opposed to a variable (named `needle` or not) in general. Just for booleans it was making me think harder than I wanted to lol